podcasts/PlaybackManager.swift

```diff
@@ -814,7 +814,12 @@ class PlaybackManager: ServerPlaybackDelegate {
 
     func playbackDidFail(logMessage: String?, userMessage: String?) {
         FileLog.shared.addMessage("playbackDidFail: \(logMessage ?? "No error provided")")
-        AnalyticsPlaybackHelper.shared.currentSource = .playbackFailed
+
+        // If there is no current analytics source, then use the previous one
+        // This helps prevent an `unknown` from being used if this is called right after another event, such as playbackPlay
+        if analyticsPlaybackHelper.currentSource == nil || analyticsPlaybackHelper.currentSource == .unknown {
```

Comment:
This is a property comparison and an assignment for `analyticsPlaybackHelper` — as far as I can see nothing here is `PlaybackManager` specific.

I wonder if we can somehow move this to `AnalyticsPlaybackHelper`?

Reply:
Done here:
- 7e7730d3d5aaf3a949de726fb20d8d07edfeeabd
- e04fc56d171df61b946c72ef16b4b2ad2761fe14